restcomm.android.client.sdk/src/main/java/org/mobicents/restcomm/android/client/sdk/RCConnection.java

```diff
@@ -560,10 +563,19 @@ void initializeWebrtc(boolean videoEnabled)
         for (String permission : MANDATORY_PERMISSIONS) {
             if (context.checkCallingOrSelfPermission(permission) != PackageManager.PERMISSION_GRANTED) {
                 logAndToast("Permission " + permission + " is not granted");
-                // TODO: return error to RCConnection listener
+                listener.onPermissionNotGranted(this, permission);
```

Comment:
I also think that RCConnection.connect() should return null when this fails (permission error, or some other error) - what do you think? Right now the onDisconnected() would be called, but the connect() still returns a non-null object, so the application will see non-null object and use that, because onDisconnected() would be executed before connect() is finished, correct?